### PR TITLE
TYPE: fold doc comments

### PR DESCRIPTION
**Pull request by vasily-kirichenko**

### CODE REVIEW

**Comment by matklad** [comment]
Great! Let's also handle INNER_EOL_DOC_COMMENT and add test as well ;)

There's org.rust.ide.folding.RsFoldingTest for it, and it had a `testBlockComment` method already. So let's rename the method into `testComments`, rename `test/resources/org/rust/ide/folding/fixtures/block_comment.rs` file into `test/resources/org/rust/ide/folding/fixtures/commenst.rs` and make it's contents more flavorful, something like

```
/*
hello
*/

///
/// Otter doc
///
fn foo() {}

mod m {
    //!
    //! Inner doc
    //! 
}
```,

with `<fold text='/* ... */'></fold>` tags enclosing the folds.

**Comment by matklad** [comment]
Oh, while we are at it, could you rename `RsFoldingTest` into `RsFoldingBuilderTest` as well, so that **Goto test** (`ctrl+shift+t`) action works in the IDE?

File: src/main/kotlin/org/rust/ide/folding/RsFoldingBuilder.kt [hunk 1]
```diff
@@ -129,7 +131,8 @@ class RsFoldingBuilder : FoldingBuilderEx(), DumbAware {
     }
 
     override fun isCollapsedByDefault(node: ASTNode): Boolean =
-        RsCodeFoldingSettings.instance.collapsibleOneLineMethods && node.elementType in COLLAPSED_BY_DEFAULT
+        (RsCodeFoldingSettings.instance.collapsibleOneLineMethods && node.elementType in COLLAPSED_BY_DEFAULT)
+            || (CodeFoldingSettings.getInstance().COLLAPSE_DOC_COMMENTS && node.elementType == OUTER_EOL_DOC_COMMENT)
```
Comment (matklad) [inline src/main/kotlin/org/rust/ide/folding/RsFoldingBuilder.kt]: :+1:, didn't think of that!

**Comment by vasily-kirichenko** [comment]
Done :) I'm not sure the test content is what it should be. I just copy pasted it from test failure diff.

**Comment by matklad** [comment]
Thanks!  (And especially for the GIF, the make changelogs so much lively!)

bors r+

>I'm not sure the test content is what it should be. I just copy pasted it from test failure diff.

Yeah, this is precisely how these sort of test work: you make sure that code works correctly by manually checking that it does the right thing when opening a rust file, and then you use the tests just to "freeze" this behavior.

**Comment by vasily-kirichenko** [comment]
Fixed the autoformatted test and renamed `RsFoldingTest`.

**Comment by matklad** [comment]
:+1: 

bors r+

**Pull request merged**